x-pack/plugin/logstash/src/main/java/org/elasticsearch/xpack/logstash/action/TransportGetPipelineAction.java

```diff
@@ -122,37 +141,66 @@ private void handleFailure(Exception e, ActionListener<GetPipelineResponse> list
         }
     }
 
-    private void handleSearchResponse(
+    private void handleFilteringSearchResponse(
         SearchResponse searchResponse,
         Map<String, BytesReference> pipelineSources,
+        Set<String> explicitPipelineIds,
+        Set<String> wildcardPipelineIdPatterns,
+        int maxNumHitsToFetch,
         Consumer<SearchResponse> clearScroll,
         ActionListener<GetPipelineResponse> listener
     ) {
+        maxNumHitsToFetch += searchResponse.getHits().getHits().length;
         for (SearchHit hit : searchResponse.getHits().getHits()) {
-            pipelineSources.put(hit.getId(), hit.getSourceRef());
+            if (explicitPipelineIds.isEmpty() && wildcardPipelineIdPatterns.isEmpty()) {
+                pipelineSources.put(hit.getId(), hit.getSourceRef());
+                continue;
+            }
+
+            // take if ID is in request IDs set
+            if (explicitPipelineIds.contains(hit.getId())) {
+                pipelineSources.put(hit.getId(), hit.getSourceRef());
+                continue;
+            }
+            // take if id matches request wildcard pattern
+            for (String wildcardPipelineIdPattern : wildcardPipelineIdPatterns) {
```

Comment:
Readability nit: You could simplify this to avoid needing the break if you changed this to a stream call:


(Assuming that `wildcardPipeilneIdPatterns` is converted to a `List<Pattern>`)
```java
if (wildcardPipelineIdPatterns.stream().anyMatch(pattern -> pattern.matcher(hit.getId()).matches())) {
        // Add entry to result map
}
```